runtime/nvqir/CircuitSimulator.h

```diff
@@ -1163,6 +1172,57 @@ class CircuitSimulatorBase : public CircuitSimulator {
     // Return the result
     return measureResult;
   }
+
+  cudaq::SpinMeasureResult measure(const cudaq::spin_op &op) override {
+    flushGateQueue();
+
+    if (executionContext->canHandleObserve) {
+      auto result = observe(*executionContext->spin.value());
+      return {*result.expectationValue, cudaq::sample_result(result)};
+    }
+
+    assert(op.num_terms() == 1 && "Number of terms is not 1.");
+
+    cudaq::info("Measure {}", op.to_string(false));
```

Comment:
If this is called in a loop, then it would probably be worth adding this to save throughput.
```suggestion
    if (cudaq::details::should_log(cudaq::details::LogLevel::info))
      cudaq::info("Measure {}", op.to_string(false));
```